### DISCUSSION TITLE
Error sending verify_email InvalidUser

### DISCUSSION THREAD

**Discussion by jalbstmeijer** [dani-garcia/vaultwarden#2244] [Q&A]
### Subject of the issue

Outgoing VERIFY EMAIL  not working. Whereby gui does not detect any issue and log is not helping pinpoint the issue even with SMTP_DEBUG enabled.

The outgoing smtp server allows relaying mail based on ip of the vaultwarden server.

### Deployment environment

* Vaultwarden version: v1.23.1
* Web-vault version: v2.25.0
* Running within Docker: true (Base: Debian)
* Environment settings overridden: false
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.35.4
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** 

```json
{
  "_duo_akey": null,
  "_enable_duo": false,
  "_enable_email_2fa": true,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_ip_header_enabled": true,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_max_conns": 10,
  "database_url": "****/**.*******",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://**.********.**",
  "domain_origin": "*****://**.********.**",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 5 * * * *",
  "emergency_request_timeout_schedule": "0 5 * * * *",
  "enable_db_wal": true,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": "/data/bitwarden.log",
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "org_attachment_limit": null,
  "org_creation_users": "",
  "password_iterations": 100000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": false,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": true,
  "smtp_explicit_tls": false,
  "smtp_from": "\"***********@********.**\"",
  "smtp_from_name": "\"VaultWarden\"",
  "smtp_host": "\"***.***.*.**\"",
  "smtp_password": null,
  "smtp_port": 25,
  "smtp_ssl": false,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "trash_auto_delete_days": null,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": "43734",
  "yubico_secret_key": "***",
  "yubico_server": null
}
```
</details>

* Install method: Docker image,
* Clients used: Chrome browser

### Steps to reproduce

Click on "Send email" of VERIFY EMAIL

### Expected behaviour

Receive a verification mail or a gui error.

### Actual behaviour

No error on gui, only this message in the log;

[WARNING] SMTP Debugging is enabled (SMTP_DEBUG=true). Sensitive information could be disclosed via logs!
[WARNING] Only enable SMTP_DEBUG during troubleshooting!

[2022-01-19 11:06:47.937][request][INFO] POST /api/accounts/verify-email
[2022-01-19 11:06:47.940][vaultwarden::api::core::accounts][ERROR] Error sending verify_email email: Address. [CAUSE] InvalidUser
[2022-01-19 11:06:47.940][response][INFO] POST /api/accounts/verify-email (post_verify_email) => 200 OK

**Comment by BlackDex** [comment]
Looks like the email is using characters not allowed, that is what this error is mentioning.
It's using this regex by the lettre (mail) library: https://github.com/lettre/lettre/blob/master/src/address/types.rs#L61-L62

**Comment by jalbstmeijer** [comment]
Ok, so with "InvalidUser",  "Invalid Email" is meant?

There is nothing special with the email address used. I have 2 accounts, both so the same.
It has always worked before, till it stopped after upgrading.

Maybe something to do with the upgrade from bitwarden_rs 1.20.0 to vaultwarden 1.23.1?

Anything I can do to debug?

**Reply by BlackDex** [reply]
There has been a lot changed between those versions, but looking at the specific code from the library it self, that part has changed for over 2 years. So, there must be something strange with the mail address.

Would be able to share the part in-front of the `@` so that we can test our selfs?

**Comment by jalbstmeijer** [comment]
sure, justin@

**Reply by BlackDex** [reply]
Well, it looks like there is something wrong in the database.
Either its having some whitespace prefixed or postfixed to it.
Or something is wrong with the sqlite database, but that is hard to tell.

**Comment by jalbstmeijer** [comment]
Anything I could try to debug further? Could I request the raw value somewhere?

Two things I can imagine to do;
- create a new user and see if that one works
- change email to temporary one and than back to the original one and see if it works.

**Reply by jalbstmeijer** [reply]
Found it, it was not my users email address causing issue but the sender email address. (SMTP_FROM)

The issue was that I was using quotes in my docker-compose.yml file for the environment values.

So maybe good to evaluate these environment values on start and or be more specific which email value is causing issues in the log.

Also nice if the gui would indicate an issue when such error occurs.

**Reply by dumblob** [reply]
@jalbstmeijer yeah, yet again YAML strikes back. I so much dislike the format. Yet it became so surprisingly ubiquitous :cry:.

**Comment by rajeshisnepali** [comment]
For me I had `SMTP_FROM="noreply@mydomain.com"`   
which I changed it to `SMTP_FROM=noreply@mydomain.com`

just removed the quotes `"`. 

.env
```
SMTP_HOST=in-v3.mailjet.com
SMTP_USERNAME=c279536d84d14b805b573
SMTP_PASSWORD=218a66c9c758bfccb6a42d
SMTP_PORT=587
SMTP_SECURITY=starttls
SMTP_FROM=noreply@mydomain.com
```

**Comment by ln-12** [comment]
Hey, did anyone find a way to use a value for SMTP_FROM in the form of `Vaultwarden <noreply@domain.com>`?

Edit: Nevermind, Vaultwarden automatically added it.
